www/src/pages/editor/NewMineralSiteModal.tsx

```diff
@@ -0,0 +1,384 @@
+      );
+    });
+    unitStore.fetchAll().then(() => {
+      setUnitOptions(
+        Array.from(unitStore.records.values())
+          .filter((unit): unit is Unit => unit !== null)
+          .map((unit) => ({ value: unit.id ?? "", label: unit.name ?? "Unnamed" }))
+      );
+    });
+    const generatedId = `record-${uuidv4()}`;
+    setRecordId(generatedId);
+    form.setFieldsValue({ recordId: generatedId });
+  }, [commodityStore, countryStore, stateOrProvinceStore, form, visible]);
+
+  const handleSourceTypeChange = (e: RadioChangeEvent) => {
+    const value = e.target.value;
+    setSelectedSourceType(value);
+
+    const refDocUrl = form.getFieldValue("refDoc");
+    const currentUser = userStore.getCurrentUser()?.url;
+
+    if (value === "unpublished") {
+      form.setFieldsValue({ sourceId: `unpublished::${currentUser || "unknown"}` });
+    } else if (refDocUrl) {
+      form.setFieldsValue({ sourceId: `${value}::${refDocUrl}` });
+    } else {
+      form.setFieldsValue({ sourceId: null });
+    }
+  };
+
+  const handleSave = async (values: any) => {
+    try {
+      let location = undefined;
+      if (values.latitude !== undefined && values.longitude !== undefined) {
+        location = `POINT (${values.longitude} ${values.latitude})`;
+      }
+
+      const countries = values.country
+        ? [
+            new CandidateEntity({
+              observedName: values.country,
+              source: values.country,
+              normalizedURI: userStore.getCurrentUser()?.url,
+              confidence: 1.0,
+            }),
+          ]
+        : [];
+
+      const statesOrProvinces = values.stateorprovince
+        ? [
+            new CandidateEntity({
+              observedName: values.stateorprovince,
+              source: values.stateorprovince,
+              normalizedURI: userStore.getCurrentUser()?.url,
+              confidence: 1.0,
+            }),
+          ]
+        : [];
+
+      const commodity1 = commodity.id;
+      console.log("COMMOdity", commodity1);
+      const referenceDocument = new Document({
+        uri: values.refDoc,
+        title: values.refDoc,
+      });
+
+      const reference = new Reference({
+        document: referenceDocument,
+        comment: values.refComment || "",
+      });
+      const sourceType = values.sourceType;
+      const refDocUrl = values.refDoc;
+      let combinedSourceId = " ";
+
+      if (sourceType === "unpublished") {
+        combinedSourceId = `unpublished::${userStore.getCurrentUser()?.url || "unknown"}`;
+      } else if (sourceType && refDocUrl) {
+        combinedSourceId = `${sourceType}::${refDocUrl}`;
+      } else {
+        combinedSourceId = " ";
+      }
+
+      if (!combinedSourceId) {
+        console.error("Error: Source ID could not be constructed. Check sourceType and refDoc values.");
+      }
+
+      const currentUser = userStore.getCurrentUser()?.name;
+      const createdBy = `https://minmod.isi.edu/users/u/${currentUser}`;
```

Comment:
use something like this instead of hard code -- user.getCurrentUser().url

Reply:
userStore.getCurrentUser()?.url this is undefined